File: my-solutions/0243-shortest-word-distance/solution.scala
object Solution {
    def shortestDistance(words: Array[String], word1: String, word2: String): Int = {
        var wordOnePointer:Int = -1
        var wordTwoPointer:Int = -1
        var minDistance = words.length

        for(el <- 0 until words.length)
        {
            if(words(el).equalsIgnoreCase(word1))
                wordOnePointer = el
            if(words(el).equalsIgnoreCase((word2)))
                wordTwoPointer = el

            if(wordOnePointer != -1 && wordTwoPointer != -1){
                minDistance = Math.min(minDistance, Math.abs(wordOnePointer - wordTwoPointer))
            }
        }
        minDistance 
    }
}

File: my-solutions/0243-shortest-word-distance/solution.py
class Solution:
    def shortestDistance(self, wordsDict: List[str], word1: str, word2: str) -> int:
        word_1_index = -1
        word_2_index = -1

        min_distance = math.inf

        for i in range(len(wordsDict)):

            if wordsDict[i] == word1:
                word_1_index = i

            if wordsDict[i] == word2:
                word_2_index = i

            if word_1_index != -1 and word_2_index != -1:
                min_distance = min(min_distance, abs(word_1_index - word_2_index))
            
        return min_distance
        
        
